airflow/config_templates/config.yml

```diff
@@ -1416,12 +1419,77 @@ api:
       description: |
         Number in seconds until the JWT token used for authentication expires for CLI commands.
         When the token expires, all CLI calls using this token will fail on authentication.
+
         Make sure that time on ALL the machines that you run airflow components on is synchronized
         (for example using ntpd) otherwise you might get "forbidden" errors.
       version_added: 3.0.0
       type: integer
       example: ~
       default: "3600"
+api_auth:
```

Comment:
I decided to move the auth-related config to its own section, as there was starting to be a longer list of them, and since we don't have nested config sections this seemed like the next best option.

If anyone feels strongly the otherway I can move these back under `api` section

Reply:
I don't have a strong opinion on moving it under `api_auth`. It looks good. The separate heading would be more memorizable. Thanks!
Shouldn't we move `auth_jwt_expiration_time` and `auth_jwt_cli_expiration_time` since they relate to `api_auth`? Maybe not in this PR, though, since they are not directly related to this piece of work. I am asking out of curiosity. I can follow up with that piece of work too.

Reply:
I am fine with that too. to be honest I dont feel strongly between `api` and `api_auth`, I can see both arguments. If we go with `api_auth`, we definitely should move `auth_jwt_expiration_time` and `auth_jwt_cli_expiration_time` as well

Reply:
Do we need a different expiration period for CLI and none CLI use? Is there any practical difference in the token?

Reply:
I change my mind on this one -- its a lot of churn to move it all around.